WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewModel.swift

```diff
@@ -56,6 +57,22 @@ final class CardReaderSettingsConnectedViewModel: CardReaderSettingsPresentedVie
         connectedReaderBatteryLevel = String.localizedStringWithFormat(Localization.batteryLabelFormat, batteryLevelString)
     }
 
+    /// Dispatch a request to check for reader updates
+    ///
+    func checkForCardReaderUpdate() {
+        let action = CardPresentPaymentAction.checkForCardReaderUpdate() { [weak self] result in
+            switch result {
+            case .success(let update):
+                self?.readerUpdateAvailable = update != nil ? .isTrue : .isFalse
```

Comment:
Your choice, but guarding against update being nil would read a bit "swiftier".

Reply:
I had it that way first, but it was so much more verbose, i.e.:

```
guard update != nil else {
    self?.readerUpdateAvailable = .isFalse
    self?.didUpdate?()
    return
}

self?.readerUpdateAvailable = .isTrue
self?.didUpdate?()
```

I'd like to leave the ternary. But I do see the opportunity to guard early and return for nil self and I will do that